components/layout/animation.rs

```diff
@@ -30,19 +30,49 @@ pub fn update_animation_state(constellation_chan: &IpcSender<ConstellationMsg>,
     let mut new_running_animations = vec![];
     while let Ok(animation) = new_animations_receiver.try_recv() {
         let mut should_push = true;
-        if let Animation::Keyframes(ref node, ref name, ref state) = animation {
-            // If the animation was already present in the list for the
-            // node, just update its state, else push the new animation to
-            // run.
-            if let Some(ref mut animations) = running_animations.get_mut(node) {
-                // TODO: This being linear is probably not optimal.
-                for mut anim in animations.iter_mut() {
-                    if let Animation::Keyframes(_, ref anim_name, ref mut anim_state) = *anim {
-                        if *name == *anim_name {
-                            debug!("update_animation_state: Found other animation {}", name);
-                            anim_state.update_from_other(&state, timer);
-                            should_push = false;
-                            break;
+        // If the animation was already present in the list for the
+        // node, just update its state, else push the new animation to
+        // run.
+        match animation {
+            Animation::Keyframes(ref node, ref name, ref state) => {
+                if let Some(ref mut animations) = running_animations.get_mut(node) {
```

Comment:
After reading this code, I've filed #14419, because these loop-and-match things seem unnecessary.